ChanThreadWatch.csproj

```diff
@@ -10,7 +10,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>JDP</RootNamespace>
     <AssemblyName>ChanThreadWatch</AssemblyName>
-    <TargetFrameworkVersion>v2.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7.1</TargetFrameworkVersion>
```

Comment:
Looks like there's a bug\missing features in Mono's implementation of .NET 2.0. System.Threading.SemaphoreFullException does not exist in .NET 2.0, but it exists in later versions.

Reply:
The app won't run on Linux and OSX out of the box due to mono lacking System.Threading.SemaphoreFullException. What do you think on working around those architecture bugs? Or do we mostly want to keep them out?

Reply:
It was my understanding that people were already using CTW on Linux through Mono for years now, so what has changed?

Reply:
I've tried it on OSX, Windows, Arch, and Ubuntu. Looks like it has already been raised as an issue to the guys at mono. 

https://github.com/mono/mono/issues/7450

I'll close this as it seems to just be a bug with how mono handles the .Net 2.0 framework.

Reply:
This has been a long standing bug and I've been using older version of CTW on Linux while waiting for fix to semaphore crash.

Reply:
I just tested this on Windows with Mono 5.16.0.179 and CTW crashes with this exception every time a watcher is started (if all your watchers are stopped and you don't add any more, it does not crash). However, this doesn't seem to have anything to do with CTW version as I tested with multiple versions including v1.7.5 (first version of this fork) and v1.14.0 (the one you are using), and every one of them exhibited the problem.

It seems to me that this is a relatively recent regression in the new versions of Mono, as I recall CTW running fine in those circumstances (granted, that was a couple years ago).

Reply:
Regardles of which version of Mono I'm running, I've been unable to use newer versions of CTW than 1.14.0 which doesn't in my case suffer from these issues. I'm using Fedora 28 at the moment and the version of mono which ships with it and neither TypeLoadException error nor SemaphoreFullException error are problems with that particular version. I should probably fill in bug reports of both separately so we can address these issues.

Reply:
I've only really bothered until recently - I would usually just pull the source and compile it, hence it just working (It would compile in 4.7.1, where the SemaphoreFullException is not an issue). I only noticed things were off when I tried to grab the exe for my laptop from the downloads. Which version of mono are you using on Fedora?

Reply:
Fedora seems to load up different versions of mono than the latest. With Fedora 29 It gave me 4.8.0, which could mean at some point there's a regression and Fedora 28 might ship with that old version.